### PR TITLE
chore: enable strict ts for backend

### LINKED ISSUE #1015: [Chore]: enable strict typescript compiler option on backend
For some reason, `strict` is enabled for all projects except the backend.
`strict` helps catch a lot of bugs in compile time.
Right now the `120` errors are reported when `strict` is enabled on the backend.

### PR DESCRIPTION
**Pull request by khaledmashaly**
## What does this PR do?

Fixes #1015

### CODE REVIEW

File: packages/backend/src/app/authentication/authentication.service.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const authenticationService = {
                     projectId: project.id,
                 },
             })
-            user.password = undefined
+            user.password = ''
```
Comment (abuaboud) [inline packages/backend/src/app/authentication/authentication.service.ts]: Please add todo to make it undefined so api response doesn't return empty string

File: packages/backend/src/app/flows/flow-version/flow-version-entity.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { EntitySchema } from 'typeorm'
 import { Flow, FlowVersion } from '@activepieces/shared'
 import { ApIdSchema, BaseColumnSchemaPart } from '../../helper/base-entity'
 
-type FlowVersionSchema = {
+export type FlowVersionSchema = {
```
Comment (abuaboud) [inline packages/backend/src/app/flows/flow-version/flow-version-entity.ts]: You can use the FlowVersion interface in the repo and keeps this one private

File: packages/backend/src/app/authentication/authentication.service.ts (hunk 1 above)
Comment (khaledmashaly) [reply packages/backend/src/app/authentication/authentication.service.ts]: fixed in f9e85866

File: packages/backend/src/app/flows/flow-version/flow-version-entity.ts (hunk 2 above)
Comment (khaledmashaly) [reply packages/backend/src/app/flows/flow-version/flow-version-entity.ts]: fixed in b7bed4a7

**Pull request merged**